src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -113,13 +113,26 @@ private static void extractIEs(
 		}
 		JsonArray iesJsonArray = iesJsonElement.getAsJsonArray();
 		InformationElements ieContainer = new InformationElements();
-		for (JsonElement ie : iesJsonArray) {
-			JsonElement typeElement = ie.getAsJsonObject().get("type");
-			if (typeElement == null) {
+		for (JsonElement ieJsonElement : iesJsonArray) {
+			JsonElement typeElement =
+				ieJsonElement.getAsJsonObject().get("type");
+			if (typeElement == null) { // shouldn't happen
 				continue;
 			}
-			JsonObject contents =
-				ie.getAsJsonObject().get(IE_CONTENT_FIELD_KEY).getAsJsonObject();
+			if (!ieJsonElement.isJsonObject()) {
+				// the IEs we are interested in are Json objects
+				continue;
+			}
+			JsonObject ie = ieJsonElement.getAsJsonObject();
+			if (!ie.has(IE_CONTENT_FIELD_KEY)) {
+				// the IEs we are interested in have the content field
+				continue;
+			}
+			JsonElement contentsJsonElement = ie.get(IE_CONTENT_FIELD_KEY);
```

Comment:
nit: we can just call get and then check if it's null to see if the field exists

Reply:
Yes, I should definitely do that.